PIL/ImageDraw.py

```diff
@@ -366,3 +370,10 @@ def floodfill(image, xy, value, border=None):
                             pixel[s, t] = value
                             newedge.append((s, t))
             edge = newedge
+
+  
+def _color_diff(rgb1, rgb2):
+    """
+    Uses distance formula to calculate difference between two rgb values.
+    """
+    return ((rgb1[0]-rgb2[0])**2 + (rgb1[1]-rgb2[1])**2 + (rgb1[2]-rgb2[2])**2)**.5
```

Comment:
That optimization makes sense, I think it would make the code a bit more confusing though.  The linear distance you talk about sounds like it would make sense to use for speed and cleanness, but I'm not exactly sure what formula you're referring to.  But from my personal testing, speed hasn't been too big of an issue with this code.